data/datasets/folk_rnn.py

```diff
@@ -30,10 +31,26 @@ def __init__(
         if data_type == "tokenized_ABC":
             self.data_file = os.path.join(self.root, "train", "data_v2.txt")
             with open(self.data_file, "r", encoding="utf-8") as file:
-                self.data = np.array(file.read().split("\n\n"))
+                full_data = np.array(file.read().split("\n\n"))
+                
+            self.data = self.split_data(full_data, split)
+
         else:
             raise ValueError(f"{data_type} is not an allowed data type for this dataset")
 
+    def split_data(self, data: np.ndarray, split: str) -> np.ndarray:
```

Comment:
This whole function looks generic enough. Maybe move it to the base class. Also no need to pass `split` argument (it is already set so use `self.split instead` i think, unless we really want to leave this method here. Also also if we move it to base class then maybe we don't even need to pass `data` as an argument and just assume that we just split `self.data`